src/runner.ts

```diff
@@ -31,9 +48,121 @@ export class ActionRunner {
     return configFile;
   }
 
+  /**
+   * The action evaluates if the rules requirements are meet for a PR
+   * @returns a true/false statement if the rule failed. This WILL BE CHANGED for an object with information (see issue #26)
+   */
+  async validatePullRequest({ rules }: ConfigurationFile): Promise<boolean> {
+    for (const rule of rules) {
+      // We get all the files that were modified and match the rules condition
+      const files = await this.listFilesThatMatchRuleCondition(rule);
+      // We check if there are any matches
+      if (files.length === 0) {
+        this.logger.debug(`Skipping rule ${rule.name} as no condition matched`);
+        // If there are no matches, we simply skip the check
+        continue;
+      }
+      if (rule.type === "basic") {
+        const [result, missingData] = await this.evaluateCondition(rule);
+        if (!result) {
+          this.logger.error(`Missing the reviews from ${JSON.stringify(missingData.missingUsers)}`);
+          return false;
+        }
+      }
+    }
+
+    // TODO: Convert this into a list of users/teams missing and convert the output into a nice summary object -> Issue #26
+    return true;
+  }
+
+  /** Evaluates if the required reviews for a condition have been meet
+   * @param rule Every rule check has this values which consist on the min required approvals and the reviewers.
+   * @returns a [bool, error data] tuple which evaluates if the condition (not the rule itself) has fulfilled the requirements
+   * @see-also ReviewError
+   */
+  async evaluateCondition(rule: { min_approvals: number } & Reviewers): Promise<ReviewError> {
+    // This is a list of all the users that need to approve a PR
+    const requiredUsers: string[] = [];
+    // If team is set, we fetch the members of such team
+    if (rule.teams) {
+      for (const team of rule.teams) {
+        const members = await this.teamApi.getTeamMembers(team);
+        for (const member of members) {
+          // simple check to stop us from having duplicates
+          if (requiredUsers.indexOf(member) < 0) {
+            requiredUsers.push(member);
+          }
+        }
+      }
+      // If, instead, users are set, we simply push them to the array as we don't need to scan a team
+    } else if (rule.users) {
+      requiredUsers.push(...rule.users);
```

Comment:
I'm curious about what would be more expected from a user standpoint:
- put either teams or users per rule?
- put team and users in a single rule?

If we get a list of required users from a team, could we have team -> users and rule.users merged technically? or something that I don't see might prevent that?

Reply:
Judging by the code, [it's one or another](https://github.com/paritytech/review-bot/blob/main/src/file/validator.ts#L42)

Reply:
I understood it wrong and put that validation. It should be teams, or users, or teams **and** users.

I created a new issue to update it: https://github.com/paritytech/review-bot/issues/28 but I will do it in a different PR because if I modify that code I'll have conflicts with #25.